File: app/src/main/java/com/wt/apkinfo/dialog/InfoListDialog.java
package com.wt.apkinfo.dialog;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.afollestad.materialdialogs.MaterialDialog;
import com.wt.apkinfo.R;
import com.wt.apkinfo.R2;
import com.wt.apkinfo.entity.ComponentInfo;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by kenumir on 24.09.2017.
 *
 */

public class InfoListDialog extends DialogFragment {

	private static final String KEY_TITLE = "title";

	public interface OnGetData {
		List<ComponentInfo> onGetData();
	}

	public static InfoListDialog newInstance(String title) {
		InfoListDialog d = new InfoListDialog();
		Bundle args = new Bundle();
		args.putString(KEY_TITLE, title);
		d.setArguments(args);
		return d;
	}

	public InfoListDialog() {}

	private OnGetData mOnGetData;

	@Override
	public void onAttach(Context context) {
		super.onAttach(context);
		mOnGetData = (OnGetData) context;
	}

	@NonNull
	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState) {
		String title = getArguments() != null ? getArguments().getString(KEY_TITLE, "List") : "List";
		MaterialDialog dialog = new MaterialDialog.Builder(getActivity())
				.title(title)
				.positiveText(R.string.label_close)
				.adapter(new ListItemsAdapter(mOnGetData.onGetData()), new LinearLayoutManager(getActivity()))
				.build();
		RecyclerView recycler = dialog.getRecyclerView();
		recycler.addItemDecoration(new DividerItemDecoration(getActivity(), DividerItemDecoration.VERTICAL));
		return dialog;
	}

	public static class ListItemHolder extends RecyclerView.ViewHolder {

		@BindView(R2.id.text1) TextView text1;
		@BindView(R2.id.text2) TextView text2;

		public ListItemHolder(View itemView) {
			super(itemView);
			ButterKnife.bind(this, itemView);
		}
	}

	private static class ListItemsAdapter extends RecyclerView.Adapter<ListItemHolder> {
		List<ComponentInfo> data;
		public ListItemsAdapter(List<ComponentInfo> d) {
			data = d;
		}

		@NonNull
		@Override
		public ListItemHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
			return new ListItemHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_component_info, parent, false));
		}

		@Override
		public void onBindViewHolder(@NonNull ListItemHolder holder, int position) {
			ComponentInfo d = data.get(position);
			holder.text1.setText(d.name);
			holder.text2.setText(d.className);
		}

		@Override
		public int getItemCount() {
			return data != null ? data.size() : 0;
		}
	}

}

File: app/src/main/java/com/wt/apkinfo/dialog/InfoListDialog.kt
package com.wt.apkinfo.dialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.afollestad.materialdialogs.MaterialDialog
import com.wt.apkinfo.R
import com.wt.apkinfo.entity.ComponentInfo

/**
 * Created by kenumir on 24.09.2017.
 *
 */

class InfoListDialog : DialogFragment() {

    private var mOnGetData: OnGetData? = null

    interface OnGetData {
        fun onGetData(): List<ComponentInfo>
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mOnGetData = context as OnGetData
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val title = if (arguments != null) arguments!!.getString(KEY_TITLE, "List") else "List"
        val dialog = MaterialDialog.Builder(this.activity!!)
                .title(title)
                .positiveText(R.string.label_close)
                .adapter(ListItemsAdapter(mOnGetData!!.onGetData()), LinearLayoutManager(this.activity!!))
                .build()

        //val dialog = MaterialDialog(this.activity!!)
        //        .title(0, title)
        //        .positiveButton(R.string.label_close)
        //        .customListAdapter(ListItemsAdapter(mOnGetData!!.onGetData()))
        //dialog.setTitle(title)
        dialog.getRecyclerView().addItemDecoration(DividerItemDecoration(activity!!, DividerItemDecoration.VERTICAL))
        return dialog
    }

    class ListItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        var text1: TextView
        var text2: TextView

        init {
            text1 = itemView.findViewById(R.id.text1)
            text2 = itemView.findViewById(R.id.text2)
        }
    }

    private class ListItemsAdapter(internal var data: List<ComponentInfo>?) : RecyclerView.Adapter<ListItemHolder>() {

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListItemHolder {
            return ListItemHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_component_info, parent, false))
        }

        override fun onBindViewHolder(holder: ListItemHolder, position: Int) {
            val d = data!![position]
            holder.text1.text = d.name
            holder.text2.text = d.className
        }

        override fun getItemCount(): Int {
            return if (data != null) data!!.size else 0
        }
    }

    companion object {

        private val KEY_TITLE = "title"

        fun newInstance(title: String): InfoListDialog {
            val d = InfoListDialog()
            val args = Bundle()
            args.putString(KEY_TITLE, title)
            d.arguments = args
            return d
        }
    }

}
